deploy/lib/xquery/setup.xqy

```diff
@@ -2417,41 +2365,42 @@ declare function setup:remove-existing-geospatial-element-indexes(
 
 declare function setup:validate-range-field-indexes($admin-config, $database, $db-config)
 {
-  try
-  {
-    let $existing :=
-      xdmp:eval('
-        import module namespace admin = "http://marklogic.com/xdmp/admin" at "/MarkLogic/admin.xqy";
-
-        declare namespace db="http://marklogic.com/xdmp/database";
-
-        declare variable $admin-config external;
-        declare variable $database external;
-
-        admin:database-get-range-field-indexes($admin-config, $database)',
-        (xs:QName("admin-config"), $admin-config,
-         xs:QName("database"), $database))
-    for $expected in $db-config/db:range-field-indexes/db:range-field-index
-    return
-      if ($existing[fn:deep-equal(., $expected)]) then ()
-      else
-        setup:validation-fail(fn:concat("Database mismatched range field index: ", $expected/db:field-name))
-  }
-  catch($ex)
-  {
-    if ($ex/error:code = "XDMP-UNDFUN") then $admin-config
+  let $existing := admin:database-get-range-field-indexes($admin-config, $database)
+  for $expected in $db-config/db:range-field-indexes/db:range-field-index
+  return
+    if ($existing[fn:deep-equal(., $expected)]) then ()
```

Comment:
I like that calling admin:database-range-field-indexes protects from comments and ordering differences, but I see a different problem. Consider this from ml-config:

        <range-field-index>
          <scalar-type>dateTime</scalar-type>
          <field-name>field-test</field-name>
          <range-value-positions>false</range-value-positions>
          <invalid>reject</invalid>
        </range-field-index>

I've incorrectly specified "invalid" instead of "invalid-values" as the element name. The validate code doesn't catch it, but fn:deep-equal would. Likewise, in any case where the value is optional or an empty string is okay, constructing an element won't work for the comparison, because it will allow the same mistake that the index-creating code makes. 

I've also noticed that the validate code isn't actually getting called. `setup:validate-install` calls `setup:validate-databases-indexes`, which calls `setup:validate-range-field-indexes`, but nothing calls `setup:validate-install` (seems like `setup:do-setup` should). 

In light of the first observation, I propose we keep the comparisons as they are. Based on the second, I suggest we open another issue to address that.